### PR TITLE
Storybook 6.2

### PR DESCRIPTION
**Pull request by eseliger**
Stacked on top of https://github.com/sourcegraph/sourcegraph/pull/19909 and https://github.com/sourcegraph/sourcegraph/pull/19907.

This PR upgrades Storybook to version 6.2, one step towards webpack 5 / esbuild compatibility in our codebase. (Storybook was the last _real_ blocker for this, afaik - besides upgrading `monaco-editor`)

### CODE REVIEW

**Comment by eseliger** [comment]
for some reason I'm stuck in an infinite recompile loop 😞 I'll tinker around and report back here

**Comment by eseliger** [comment]
Update: No idea what is causing this, I've tried removing all config options that aren't mandatory and it still happens for me.. Hooking into fsevents also didn't yield any interesting events being thrown, so it seems this is rather internal than fsevents triggering it 🤷‍♂️

**Comment by felixfbecker** [comment]
Maybe file a bug with Storybook? Can link to this as a repro

**Comment by eseliger** [comment]
I tried a fresh setup and it didn't happen there, so I'm not sure it's their fault.. 🤔 Will need to do some more digging

**Comment by eseliger** [comment]
I found a fix! @sourcegraph/frontend-devs this is ready for review then :)

File: package.json [hunk 1]
```diff
@@ -111,19 +111,19 @@
...
-    "@storybook/addons": "^6.1.11",
-    "@storybook/components": "^6.1.11",
-    "@storybook/core": "^6.1.11",
-    "@storybook/react": "^6.1.11",
-    "@storybook/theming": "^6.1.11",
+    "@storybook/addon-knobs": "^6.2.8",
+    "@storybook/addon-links": "^6.2.8",
+    "@storybook/addon-storyshots": "^6.2.8",
+    "@storybook/addon-storyshots-puppeteer": "^6.2.8",
+    "@storybook/addon-toolbars": "6.2.8",
```
Comment (5h1rU) [inline package.json:121]: Is this pinned version on purpose?

Comment (eseliger) [reply package.json:121]: ah no, good eye, I'll fix that

File: client/storybook/src/main.ts [hunk 2]
```diff
@@ -7,7 +7,13 @@ import { Configuration, DefinePlugin, ProgressPlugin, RuleSetUseItem, RuleSetUse
 
 const rootPath = path.resolve(__dirname, '../../../')
 const monacoEditorPaths = [path.resolve(rootPath, 'node_modules', 'monaco-editor')]
-const storiesGlob = path.resolve(rootPath, 'client/**/*.story.tsx')
+// Due to an issue with constant recompiling (https://github.com/storybookjs/storybook/issues/14342)
+// we need to make the globs more specific (`(web|shared..)` also doesn't work). Once the above issue
+// is fixed, this can be removed and watched for `client/**/*.story.tsx` again.
```
Comment (limitedmage) [inline client/storybook/src/main.ts:12]: Can we file an issue on our side to take a look at this in the future?

**Review by limitedmage — APPROVED** [review]

**Pull request merged**